kubectl-cloudflow/docker/client.go

```diff
@@ -147,53 +151,71 @@ func imageMatchesNameAndTag(image types.ImageSummary, imageNameAndTag string) bo
 	return false
 }
 
-// GetStreamletDescriptorsForImage extracts the configuration of a Cloudflow label from a docker image
+// GetStreamletDescriptorsForImageWithRegistryInspection extracts the configuration of a Cloudflow label from a docker image
 // and the image with digest in a struct
-func GetStreamletDescriptorsForImage(cli *client.Client, imageName string) (cloudflowapplication.CloudflowStreamletDescriptorsDigestPair, string, *PulledImage, error) {
-	// need this till we have a way to use docker registry API to check labels
-	pulledImage, pullError := PullImage(cli, imageName)	
-	if pullError != nil {	
-		return cloudflowapplication.CloudflowStreamletDescriptorsDigestPair{}, "", nil, fmt.Errorf("Failed to pull image %s: %s", imageName, pullError.Error())
-	}
-
-	images, err := cli.ImageList(context.Background(), types.ImageListOptions{})
+func GetStreamletDescriptorsForImageWithRegistryInspection(imageName string)(cloudflowapplication.CloudflowStreamletDescriptorsDigestPair, string, *PulledImage, error) {
+	ins := InspectOptions{Image: &ImageOptions{}}
```

Comment:
How does this work? (does it somehow find the credentials, if you pass in this empty ImageOptions struct?)

Reply:
This is the default options (nothing special eg no `dockerCertPath`) and it is empty as the underlying lib relies by default on the `docker login` to have happened before, user only needs to supply the image name there.

Reply:
I will create a factory method for default options as it is common: https://stackoverflow.com/questions/18125625/constructors-in-go

Reply:
Ok great. that's cool. What happens if they have not logged in, fails?

Reply:
Yes I tested it with aws where the credentials expired.